src/symbolic_choice.mli

```diff
@@ -1,5 +1,19 @@
 exception Assertion of Encoding.Expr.t * Thread.t
 
+module Minimalist : sig
+  include
+    Choice_intf.Complete_without_run
+      with type thread := Thread.t
+       and module V := Symbolic_value.S
+
+  type 'a res =
+    | Ok of 'a
+    | Trap of Trap.t
+    | Assert_fail
```

Comment:
Same as in the `.ml` also this is not used to build values outside of the model so this can me marked as private ?

Reply:
You mean delete `type 'a res` from the `.mli` ?
It's used in
```ocaml
val run_minimalist : 'a t -> Thread.t -> 'a res * Thread.t
```

Reply:
And I meant marking the `err` type as `private` in the `.mli`:

```ocaml
type err = private
    | Assert_fail
 ...
```

The private keywords allows other modules to match on a value of type `err` but not to create one.

Reply:
> It is simply an alias. You can delete it and replace it by its definition.

I couldn't delete `val run_minimalist` in `.mli` because it's used in `Interprets.Owi_symbolic` module.
Maybe there's a better way to do it ?

Reply:
You should delete the `res` type, not the `run_minimalist` function.